cartridges/openshift-origin-cartridge-haproxy/bin/control

```diff
@@ -175,17 +179,22 @@ function _reload_service() {
 
 function start() {
     _start_haproxy_service
-    isrunning  &&  echo "HAProxy instance is started"
+    isrunning  &&  client_result "HAProxy instance is started"
 }
 
 function stop() {
     _stop_haproxy_service
-    isrunning  ||  echo "HAProxy instance is stopped"
+    if `pgrep -x haproxy > /dev/null 2>&1`; then
+      client_error "Could not stop haproxy, use restart or application force-stop to kill."
+      exit 1
+    else
+      isrunning  || client_result "HAProxy instance is stopped"
```

Comment:
It seems like `isrunning` does approximately the same check as you are doing above: You do `pgrep -x haproxy` above whereas `isrunning` does `ps --pid $haproxy_pid` and then `pgrep -x haproxy -u $current_user`.  It's not clear to me why you cannot just use `isrunning` in `stop()`.